src/js/views/maps/LayerCategoryItemView.js

```diff
@@ -0,0 +1,203 @@
+
+'use strict';
+
+define(
+  [
+    'jquery',
+    'underscore',
+    'backbone',
+    'text!templates/maps/layer-category-item.html',
+    'models/maps/AssetCategory',
+    'common/IconUtilities',
+    // Sub-views
+    'views/maps/LayerListView',
+  ],
+  function (
+    $,
+    _,
+    Backbone,
+    Template,
+    AssetCategory,
+    IconUtilities,
+    // Sub-views
+    LayerListView,
+  ) {
+
+    /**
+    * @class LayerCategoryItemView
+    * @classdesc One item in a Category List: shows some basic information about the
+    * layer category, including label and icon. Also has a button that expands the
+    * nested layers list.
+    * @classcategory Views/Maps
+    * @name LayerCategoryItemView
+    * @extends Backbone.View
+    * @screenshot views/maps/LayerCategoryItemView.png
+    //  TODO: yvonneshi - update
+    // * @since x.x.x
+    * @constructs
+    */
+    const LayerCategoryItemView = Backbone.View.extend(
+      /** @lends LayerCategoryItemView.prototype */{
+
+        /**
+        * The type of View this is
+        * @type {string}
+        */
+        type: 'LayerCategoryItemView',
+
+        /**
+        * The HTML classes to use for this view's element
+        * @type {string}
+        */
+        className: className,
+
+        /**
+        * The model that this view uses
+        * @type {AssetCategory}
+        */
+        model: undefined,
+
+        /**
+         * The primary HTML template for this view
+         * @type {Underscore.template}
+         */
+        template: _.template(Template),
+
+        classNames: {
+          baseClass: className,
+          metadata: className + '__metadata',
+          icon: className + '__icon',
+          expanded: className + '__expanded',
+          collapsed: className + '__collapsed',
+          layers: className + '__layers',
+        },
+
+        /**
+        * A function that gives the events this view will listen to and the associated
+        * function to call.
+        * @returns {Object} Returns an object with events in the format 'event selector':
+        * 'function'
+        */
+        events() {
+          try {
+            return {[`click .${this.classNames.metadata}`]: 'toggleExpanded'};
+          }
+          catch (error) {
+            console.log(
+              'There was an error setting the events object in a LayerCategoryItemView' +
+              '. Error details: ' + error
+            );
+          }
+        },
+
+        /**
+        * Executed when a new LayerCategoryItemView is created
+        * @param {Object} options - A literal object with options to pass to the view
+        */
+        initialize(options) {
+          try {
```

Comment:
I don't think this can throw, can you remove this error handling?

Reply:
Removed